server/src/main/java/org/apache/druid/server/initialization/jetty/ChatHandlerServerModule.java

```diff
@@ -67,11 +69,21 @@ public void configure(Binder binder)
 
     Multibinder.newSetBinder(binder, ServletFilterHolder.class).addBinding().to(TaskIdResponseHeaderFilterHolder.class);
 
-    /**
-     * We bind {@link DruidNode} annotated with {@link RemoteChatHandler} to {@literal @}{@link Self} {@link DruidNode}
-     * so that same Jetty Server is used for querying as well as ingestion.
-     */
-    binder.bind(DruidNode.class).annotatedWith(RemoteChatHandler.class).to(Key.get(DruidNode.class, Self.class));
+    if (useSeparatePort) {
+      // bind a modified DruidNode that will be used by the Jetty server installed below
+      binder.bind(DruidNode.class)
```

Comment:
Would it be possible to internally use a QoS filter with jetty to leave some room for chat handler traffic instead relying on a separate server with a separate pool? I don't think this would require bubbling up to the user configs, just something we compute based on the top level configs (though it might require some minimum).

I think it would simplify things both conceptually and for the implementation, and would make this no longer an issue I think.

Reply:
I redid this to use QoS filters, the configured number of connections are doubled (one set for chat handler, the other for non-chat handler requests), with 2 reserved connections for lookups